### DISCUSSION TITLE
Where can i learn what does each tag means?

### DISCUSSION THREAD

**Discussion by MonsterMMORPG** [sergey-tihon/Stanford.NLP.NET#115] [Q&A]
I am testing Arabic POS tagger and it shows tags as

I have found here but almost all links are broken and no useful info : http://nlp.stanford.edu/software/parser-arabic-faq.shtml

اللاعب/DTNN
عامل/NN
اكتب/VBP
سجل/VBD
تسجل/VBP
قيد/NN
أرسل/VBD بالبريد/NNP المسجل/DTJJ
سجل/VBD السيارة/DTNN
دون/NN في/IN سجل/NN

Where can i learn what does each tag means? And i need the full list? Ty

This is the code i use

```
            var tagger = new MaxentTagger(@"D:\77 sozluk projesi\StandFordPosTagger\stanford-parser-full-2016-10-31\stanford-parser-3.7.0-models\edu\stanford\nlp\models\pos-tagger\arabic\arabic.tagger");

            List<string> lstTryWords = new List<string> { "البشر", "بشري", "اللاعب", "عامل", "اكتب", "سجل", " تسجل", " قيد", "أرسل بالبريد المسجل", " سجل السيارة", " دون في سجل" };

            foreach (var item in lstTryWords)
            {
                var sentences = MaxentTagger.tokenizeText(new java.io.StringReader(item.Trim())).toArray();
                foreach (java.util.ArrayList sentence in sentences)
                {
                    var taggedSentence = tagger.tagSentence(sentence);
                    Debug.WriteLine(SentenceUtils.listToString(taggedSentence, false));
                }
            }
```

**Comment by mdileep** [comment]
Following page may help you.
https://www.ling.upenn.edu/courses/Fall_2003/ling001/penn_treebank_pos.html

**Comment by sergey-tihon** [comment]
Thank you @mdileep 
Also you can look into this https://github.com/sergey-tihon/Stanford.NLP.Fsharp/blob/master/src/Stanford.NLP.Parser.Fsharp/PennTreebankIITags.fs
Some time ago I did intellisence over tags for F# users
http://sergey-tihon.github.io/Stanford.NLP.Fsharp/

**Comment by MonsterMMORPG** [comment]
Ty very much for answers. Is this list up to date?
https://www.ling.upenn.edu/courses/Fall_2003/ling001/penn_treebank_pos.html

Also i dont get some of them. For example What does Coordinating
conjunction mean?

Also lets say it is JJR

Adjective, comparative . how can i find its base adjective?

also @sergey-tihon  and @mdileep  i can not find tag DTNN in non of the links you have provided.

**Comment by sergey-tihon** [comment]
> Regarding your question (المدرسة/DTNN), they mentioned that it consists of two tags (DT + NN) where DT= (الــ) (pronounced as 'Al') and NN = (Noun, singular or mass) see Penn Treebank .

From SO - http://stackoverflow.com/a/29734148/1429493

**Comment by MonsterMMORPG** [comment]
@sergey-tihon ty for answer

but i still can not make any sense of what does each part means

for example DET+ADJ_COMP+NSUFF_MASC_DU_GEN DT+JJR

what does DET , ADJ_COMP, NSUFF_MASC_DU_GEN, DT, JJR  means?

Also is that possible to find related root words? I mean for example 

easy is adjective , easier is comparative

lets say the tagger says it is comparative form. How can i find its root adjective?

Also can we determine plural and non-plural forms? Lets say it says it is plural form. Can i get its non-plural form from tagger?

**Comment by sergey-tihon** [comment]
@MonsterMMORPG definitely better to ask question about the meaning to authors ;)

I could recommend to [play online with corenlp](http://corenlp.run
) annotations (sound like you need to add dependency parse to your pipeline):

<img width="1217" alt="screen shot 2017-03-10 at 12 30 25 pm" src="https://cloud.githubusercontent.com/assets/1197905/23789784/619fea7a-058d-11e7-87e5-73e68ca6449d.png">

**Comment by MonsterMMORPG** [comment]
@sergey-tihon the problem is i dont know arabic :D also for example for the word easier in arabic it parses as VBD. But VBD is verb past tense according to here : https://www.ling.upenn.edu/courses/Fall_2003/ling001/penn_treebank_pos.html

Where can i reach the authors? email?

**Comment by sergey-tihon** [comment]
@MonsterMMORPG I also do not know arabic ;)

From the official github repo - https://github.com/stanfordnlp/CoreNLP

> Questions about CoreNLP can either be posted on StackOverflow with the tag [stanford-nlp](http://stackoverflow.com/questions/tagged/stanford-nlp),
  or on the [mailing lists](http://nlp.stanford.edu/software/corenlp.shtml#Mail).

but twitter account is also an option https://twitter.com/stanfordnlp
+ email  java-nlp-support@lists.stanford.edu

**Comment by MonsterMMORPG** [comment]
@sergey-tihon do you know where can i learn tags like these meanings?

DET
NOUN_NUM
NSUFF_FEM_SG
CASE_DEF_ACC DT
CD

**Comment by sergey-tihon** [comment]
@MonsterMMORPG no ideas rather than official docs & source code
